samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -110,7 +112,7 @@ def prepare(params: dict) -> dict:
 
         # store in supplied output dir
         if not os.path.exists(output_dir_path):
-            os.mkdir(output_dir_path)
+            os.makedirs(output_dir_path, exist_ok=True)
```

Comment:
makedirs would create any intermediate directories too

Reply:
Looks like the gates passed, so my question is moot. 

But I will throw this out there anyway, sometimes params like exist_ok or any new params that get added are from a certain python version. So It might work locally, but will fail at the gates, just something to have in mind for the future.